src/main/java/org/jabref/JabRefMain.java

```diff
@@ -65,6 +69,25 @@ private static void start(String[] args) {
         }
 
         Globals.prefs = preferences;
+
+        // Check if we are running an acceptable version of Java
+        final BuildInfo buildInfo = Globals.BUILD_INFO;
+        if (!buildInfo.isAcceptableJavaVersion()) {
+            StringBuilder versionError = new StringBuilder(
+                    Localization.lang("Your current Java version (%0) is not supported. Please install version %1 or higher.",
+                    JavaVersionCheck.getJavaVersion(),
+                    buildInfo.getMinRequiredJavaVersion()
+                    )
+            );
+            if (!buildInfo.isAllowJava9()) {
+                versionError.append("\n");
+                versionError.append(Localization.lang("Note that currently, JabRef does not run with Java 9."));
+            }
+            final JFrame frame = new JFrame();
```

Comment:
Since we are trying to move to JavaFX, would it be possible to display the error message with a FX window instead of a `JFrame`?

Reply:
Have a look at the FXDialogService.class. there are already some predefined dialogs for fx

Reply:
@Siedlerchr @lenhard That was on purpose. I was using `JFrame` because what good will it be if I check the Java version, try to display a message and crash with an exception because FX is not available? :smile:

Can we safely assume the FXDialog will always work?

Reply:
@lenhard I made JabRef quit only when Java 9 is used. Otherwise it displays the message but continues

Reply:
@halirutan  if no openjfx is avaiable it will crash nonetheless because the JabRef main inherits from Application (javafx.application.Application). That is the thing which you can't check for directly.
The only thing would be we use a "Pre-Main-Class" that does nit depend on javafx. But I am unsure if this would work

Reply:
@Siedlerchr Ha.. overlooked this. My genius idea was to put the test at the very beginning before any of the fx stuff is instantiated. OK, I'm looking into fx dialog tonight.